Always register default keymaps

Faria22 opened the pull request:
- remove the register_keymaps toggle and make default conflict mappings install on every setup()
  - allow per-key override/disable via opts.keymaps, with tests stubbing vim.keymap.set to cover defaults and customisations
  - trim the lazy.nvim install example and document the built-in keymaps separately

Faria22 committed:
Use command calls in lazy.nvim keymap example

Faria22 committed:
feature: Handle stash conflict markers

Faria22 committed:
First iteration of quickfix support

Faria22 committed:
Quickfix support with fix for multiple repeated entrances

Faria22 committed:
Add quickfix conflict listing

Faria22 committed:
Updated configuration to be as simple as possible

StackInTheWild commented:
I'm not sure why to always register on setup. Someone told me it wasn't a good practice with Lazy. I'll have to check or perhaps you can explain

Faria22 committed:
feat: simplify key mappings configuration

Faria22 commented:
I'm not sure, I never head that that was a bad practice, and I've seen many plugins do it. I'm just updating a few things that I just noticed were broken.

Faria22 committed:
feat: add enabled toggle

Faria22 committed:
Merge branch 'main' into readme

Faria22 commented:
So what I changed is that now it uses the setup method, and I added a "enabled" boolean to toggle the plugin. Now it also makes it easier just to overwrite the default key mappings. I will be honestly, this is pretty different from what you originally have, but I think this is more in line with what other plugins use.

StackInTheWild commented:
> So what I changed is that now it uses the setup method, and I added a "enabled" boolean to toggle the plugin. Now it also makes it easier just to overwrite the default key mappings. I will be honestly, this is pretty different from what you originally have, but I think this is more in line with what other plugins use.

I'll take a look when I have time, probably in a day or two
